packages/core/src/html/siteAndPageNavProcessor.js

```diff
@@ -13,6 +14,53 @@ const {
   SITE_NAV_DROPDOWN_ICON_HTML,
 } = require('../Page/constants');
 
+/**
+ * Replaces and stores a uuid identifier to the only page-nav element, if there is one.
+ *
+ * The page-nav can only be used inside a layout,
+ * but can be constructed only after the page has been built.
+ * Hence, this requires post insertion of the page nav.
+ *
+ * This uuid identifier is asserted to be unique in the html file once html processing is done,
+ * otherwise it is replaced with one until it is unique.
+ */
+class PageNavProcessor {
+  constructor() {
+    this.uuidTextNode = undefined;
+  }
+
+  getUuid() {
+    return (this.uuidTextNode && this.uuidTextNode.data) || '';
+  }
+
+  renderPageNav(node) {
+    [this.uuidTextNode] = cheerio.parseHTML(uuidv4());
+    cheerio(node).replaceWith(this.uuidTextNode);
```

Comment:
Can I check what does this line of code actually does? :O

I think the purpose is to completely replace the node with the `uuid` right?

Reply:
yep, the `<page-nav />` wrapper itself is also replaced here with just the uuid text